subgraphs/compound-forks/rari-fuse/src/mappings.ts

```diff
@@ -0,0 +1,942 @@
+  }
+
+  // set liquidation incentive for pool entity
+  let tryLiquidationIncentive = troller.try_liquidationIncentiveMantissa();
+  if (tryLiquidationIncentive.reverted) {
+    log.warning(
+      "[getOrCreateProtocol] liquidationIncentiveMantissaResult reverted",
+      []
+    );
+  } else {
+    pool.liquidationIncentive = tryLiquidationIncentive.value
+      .toBigDecimal()
+      .div(mantissaFactorBD)
+      .times(BIGDECIMAL_HUNDRED);
+  }
+  pool.save();
+}
+
+//////////////////////////////
+//// Comptroller Handlers ////
+//////////////////////////////
+
+// Note: these are pool level functions in fuse, but each pool is a Comptroller impl
+// Source: https://docs.rari.capital/fuse
+
+// add a new market
+export function handleMarketListed(event: MarketListed): void {
+  let protocol = LendingProtocol.load(FACTORY_CONTRACT);
+  if (!protocol) {
+    // best effort
+    log.warning("[handleMarketListed] Protocol not found: {}", [
+      FACTORY_CONTRACT,
+    ]);
+    return;
+  }
+
+  // get/create ctoken
+  CTokenTemplate.create(event.params.cToken);
+  let cTokenContract = CToken.bind(event.params.cToken);
+  let cToken = new TokenData(
+    event.params.cToken,
+    getOrElse(cTokenContract.try_name(), "UNKNOWN"),
+    getOrElse(cTokenContract.try_symbol(), "UNKNOWN"),
+    getOrElse(cTokenContract.try_decimals(), -1)
+  );
+
+  // get/create underlying token
+  let underlyingAddress = getOrElse(
+    cTokenContract.try_underlying(),
+    Address.fromString(ZERO_ADDRESS)
+  );
+
+  let underlyingToken: TokenData;
+  if (underlyingAddress == Address.fromString(ZERO_ADDRESS)) {
+    // this is ETH
+    underlyingToken = new TokenData(
+      Address.fromString(ETH_ADDRESS),
+      ETH_NAME,
+      ETH_SYMBOL,
+      mantissaFactor
+    );
+  } else {
+    let underlyingContract = ERC20.bind(underlyingAddress);
+    underlyingToken = new TokenData(
+      underlyingAddress,
+      getOrElse(underlyingContract.try_name(), "UNKNOWN"),
+      getOrElse(underlyingContract.try_symbol(), "UNKOWN"),
+      getOrElse(underlyingContract.try_decimals(), -1)
+    );
+  }
+
+  // populate market data
+  let marketData = new MarketListedData(
+    protocol,
+    underlyingToken,
+    cToken,
+    getOrElse(cTokenContract.try_reserveFactorMantissa(), BIGINT_ZERO)
+  );
+
+  _handleMarketListed(marketData, event);
+
+  // fuse-specific: add fuseFees and adminFees
+
+  // get fuse fee - rari collects this (ie, protocol revenue)
+  let tryFuseFeeMantissa = cTokenContract.try_fuseFeeMantissa();
+  updateOrCreateRariFee(
+    tryFuseFeeMantissa.reverted ? BIGINT_ZERO : tryFuseFeeMantissa.value,
+    RariFee.FUSE_FEE,
+    event.params.cToken.toHexString()
+  );
+
+  // get admin fee - pool owners (admin) collect this (ie, protocol revenue)
+  let tryAdminFeeMantissa = cTokenContract.try_adminFeeMantissa();
+  updateOrCreateRariFee(
+    tryAdminFeeMantissa.reverted ? BIGINT_ZERO : tryAdminFeeMantissa.value,
+    RariFee.ADMIN_FEE,
```

Comment:
Hmm, not sure if admin fee should be considered protocol revenue. Probably should check with @pietydespises